cloud: add compression formats (#13447)
pingcap/docs: tidb-cloud/import-csv-files.md

Reviewer: The following content also needs to be updated. 
![image](https://user-images.githubusercontent.com/79440533/236731007-c988845e-8bbe-42c2-93d3-b79395edccd1.png)

Author: updated

    - If a CSV file contains all data of an entire table, name the file in the `${db_name}.${table_name}.csv` format, which maps to the `${db_name}.${table_name}` table when you import the data.
    - If the data of one table is separated into multiple CSV files, append a numeric suffix to these CSV files. For example, `${db_name}.${table_name}.000001.csv` and `${db_name}.${table_name}.000002.csv`. The numeric suffixes can be inconsecutive but must be in ascending order. You also need to add extra zeros before the number to ensure all the suffixes are in the same length.

>     - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format, in which `${suffix}` is optional and can be any number such as '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you need to rename the file as `bikeshare.trips.000001.csv.gz`.

Reviewer 2: ```suggestion
    - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format, in which `${suffix}` is optional and can be any integer such as '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you need to rename the file as `bikeshare.trips.000001.csv.gz`.
```

    - If a CSV file contains all data of an entire table, name the file in the `${db_name}.${table_name}.csv` format, which maps to the `${db_name}.${table_name}` table when you import the data.
    - If the data of one table is separated into multiple CSV files, append a numeric suffix to these CSV files. For example, `${db_name}.${table_name}.000001.csv` and `${db_name}.${table_name}.000002.csv`. The numeric suffixes can be inconsecutive but must be in ascending order. You also need to add extra zeros before the number to ensure all the suffixes are in the same length.

>     - TiDB Cloud supports importing the following compressed file formats: `.gzip`, `.gz`, `.zstd`, and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.

Author: ```suggestion
    - TiDB Cloud supports importing the following compressed file formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.csv.gz` file to the `bikeshare.trips` table, you can name the file as `bikeshare.trips.000001.csv.gz`.

Reviewer 2: ```suggestion
    - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing the following compressed file formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.

Author: ```suggestion
    - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.

Reviewer 2: ```suggestion
    - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format,${suffix} is optional and can be any number like '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you have to rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.

Author: ```suggestion
    - TiDB Cloud supports importing the following compressed file formats: `.gzip`, `.gz`, `.zstd`, and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you can rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format,${suffix} is optional and can be any number like '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you have to rename the file as `bikeshare.trips.000001.csv.gz`.

Author: ```suggestion
    - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format, in which ${suffix} is optional and can be any number such as '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you have to rename the file as `bikeshare.trips.000001.csv.gz`.
```

>     - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format, in which ${suffix} is optional and can be any number such as '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you have to rename the file as `bikeshare.trips.000001.csv.gz`.

Reviewer: ```suggestion
    - TiDB Cloud supports importing compressed files in the following formats: `.gzip`, `.gz`, `.zstd`, `.zst` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.${suffix}.csv.${compress}` format, in which `${suffix}` is optional and can be any number such as '000001'. For example, if you want to import the `trips.000001.csv.gz` file to the `bikeshare.trips` table, you need to rename the file as `bikeshare.trips.000001.csv.gz`.
```

    > **Note:**
    >

>     > - To achieve better performance, it is recommended to limit the size of each compressed file to 100 MiB.

Author: ```suggestion
    > - You only need to compress the data files, not the database and table schema files.
    > - To achieve better performance, it is recommended to limit the size of each compressed file to 100 MiB.
```

>     > - If you cannot update the CSV filenames according to the preceding rules in some cases (for example, the CSV file links are also used by your other programs), you can keep the filenames unchanged and use the **File Pattern** in [Step 4](#step-4-import-csv-files-to-tidb-cloud) to import your source data to a single target table.

Reviewer: Is `If you cannot update the CSV filenames according to the preceding rules in some cases (for example, the CSV file links are also used by your other programs), you can keep the filenames unchanged and use the **File Pattern** in [Step 4](#step-4-import-csv-files-to-tidb-cloud)` also applicable to compressed files?

Reviewer 3: Currently file pattern cannot apply on compressed files until we support it in our Web UI

Reviewer 2: No,it is not available for compressed source files. @hfxsd @qiancai

    - If a CSV file contains all data of an entire table, name the file in the `${db_name}.${table_name}.csv` format, which maps to the `${db_name}.${table_name}` table when you import the data.
    - If the data of one table is separated into multiple CSV files, append a numeric suffix to these CSV files. For example, `${db_name}.${table_name}.000001.csv` and `${db_name}.${table_name}.000002.csv`. The numeric suffixes can be inconsecutive but must be in ascending order. You also need to add extra zeros before the number to ensure all the suffixes are in the same length.

>     - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.csv.${compress}` format. For example, if you want to import the `trips.csv.gz` file to the `bikeshare.trips` table, you can name the file as `bikeshare.trips.csv.gz`.

Reviewer 3: `${db_name}.${table_name}.00000X.csv` also supports compressed files, like `bikeshare.trips.000001.csv.gz`

>     - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.csv.${compress}` format. For example, if you want to import the `trips.csv.gz` file to the `bikeshare.trips` table, you can name the file as `bikeshare.trips.csv.gz`.

Author: ```suggestion
    - TiDB Cloud supports importing the following compressed file formats: `.gz`, `.lz4`, `.zstd`, `.xz`, `.lzo` and `.snappy`. If you want to import compressed CSV files, name the files in the `${db_name}.${table_name}.000001.csv.${compress}` format. For example, if you want to import the `trips.csv.gz` file to the `bikeshare.trips` table, you can name the file as `bikeshare.trips.000001.csv.gz`.
```

    > **Note:**

>     >

Reviewer 2: To achieve better performance, it is recommended to limit the size of each compressed file to 100 MiB.

>     >

Author: ```suggestion
    >
    > To achieve better performance, it is recommended to limit the size of each compressed file to 100 MiB.
```
